pkg/generate/git/repository.go

```diff
@@ -0,0 +1,124 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+// execCmdFunc is a function that executes an external command
+type execCmdFunc func(dir, name string, args ...string) (string, string, error)
```

Comment:
Why not to just define the 'exec' function for Repository? Then you can get the directory from the struct and you don't need to have it as a first argument.